### ISSUE TITLE
Change Request: Add support --quiet flag in eslint API

### ISSUE THREAD

**Issue by pumano** [eslint/eslint#19085]
### ESLint version

9.x

### What problem do you want to solve?

I use nx monorepo and looks like they use eslint deeply by instantiating eslint as class and operate with instance. Currently eslint API not support `--quiet` flag and nx has their own `--quiet` which only filter warnings from output (as 8.x eslint if I understand correctly).

nx eslint executor code:

https://github.com/nrwl/nx/blob/9adb9e00beee38ba1f41a4a3e10a7f8a599e5b96/packages/eslint/src/executors/lint/lint.impl.ts#L176-L180

eslint cli code:

https://github.com/eslint/eslint/blob/151c965aec1c46000ac7dfc67a1c04802112aafc/lib/cli.js#L231-L235

### What do you think is the correct solution?

Would be good to have `--quiet` flag supported via API, and then nx could support it too, which is very good for consumers because we don't need to spend many minutes in CI pipeline for waiting warn rules.

### Participation

- [ ] I am willing to submit a pull request for this change.

### Additional comments

I also open issue https://github.com/nrwl/nx/issues/28291 in nx but later trying to check how it can be implemented.

**Comment by mdjermanovic** [comment]
Hi @pumano, thanks for the issue!

ESLint API already provides `options.ruleFilter` (https://eslint.org/docs/latest/integrate/nodejs-api#parameters, the Linting section) which allows for `--quiet` to not run rules set to warn, and other features that could be implemented on top of the API.

To replicate `--quiet` behavior, API consumers can do the same ESLint CLI is doing:

```js
new ESLint({ ruleFilter: rule => rule.severity === 2 })
```

**Comment by pumano** [comment]
@mdjermanovic OK, thank you for detailed answer!
